app/src/main/java/com/aguesoftguar/medalarm/addeditmedicine/AddEditMedicineFragment.java

```diff
@@ -593,18 +602,63 @@ public void onClick(View view) {
       inflater.inflate(R.menu.add_and_edit_medicine_settings, menu);
    }
 
+   @Override
+   public boolean onOptionsItemSelected(MenuItem item) {
+      switch (item.getItemId()) {
+
+         case R.id.save_medicine:
+            LayoutInflater inflater =
+               LayoutInflater.from(AddEditMedicineFragment.this.getContext());
+            View contentView = inflater.inflate(R.layout.activity_add_and_edit_medicine, null);
+
+            EditText medicineNameEditText =
+                    (EditText) contentView.findViewById(R.id.medicine_name_edittext);
+
+            String medicineName = medicineNameEditText.getText().toString();
+            String patientId = patientsAdapter.getKey(positionPatientSelected);
```

Comment:
A FC appears if the user hasn't selected a patient:
```
FATAL EXCEPTION: main Process: com.aguesoftguar.medalarm.development, PID: 13267 java.lang.ArrayIndexOutOfBoundsException: length=0; index=-1
   at java.util.ArrayList.get(ArrayList.java:413)
   at com.aguesoftguar.medalarm.addeditmedicine.PatientsAdapter.getKey(PatientsAdapter.java:96)
   at com.aguesoftguar.medalarm.addeditmedicine.AddEditMedicineFragment.onOptionsItemSelected(AddEditMedicineFragment.java:621)
```

We need to solve this issue. Maybe you could open an issue to solve this in another PR.